bundles/org.openhab.core.config.discovery.addon.process/src/main/java/org/openhab/core/config/discovery/addon/process/ProcessAddonFinder.java

```diff
@@ -46,35 +47,52 @@ public class ProcessAddonFinder extends BaseAddonFinder {
     public static final String SERVICE_NAME = SERVICE_TYPE + ADDON_SUGGESTION_FINDER;
 
     private static final String COMMAND = "command";
+    private static final String COMMAND_LINE = "commandLine";
+    private static final Set<String> SUPPORTED_PROPERTIES = Set.of(COMMAND, COMMAND_LINE);
 
     private final Logger logger = LoggerFactory.getLogger(ProcessAddonFinder.class);
 
-    // get list of running processes visible to openHAB,
-    // also tries to mitigate differences on different operating systems
-    String getProcessCommandProcess(ProcessHandle h) {
-        Optional<String> command = h.info().command();
-        if (command.isPresent()) {
-            return command.get();
-        }
-        Optional<String[]> args = h.info().arguments();
-        if (args.isEmpty()) {
-            return "";
-        }
-        String[] argsArray = args.get();
-        if (argsArray.length < 1) {
-            return "";
+    /**
+     * Private class to extract match property parameters from a {@link ProcessHandle.Info} object.
+     * Tries to mitigate differences on different operating systems.
+     */
+    protected static class ProcessInfo {
+        protected final @Nullable String command;
+        protected final @Nullable String commandLine;
+
+        /**
+         * Initializes the command and commandLine fields.
+         * If the command field is not present, it parses the first token in the command line.
+         */
+        protected ProcessInfo(ProcessHandle.Info info) {
+            commandLine = info.commandLine().orElse(null);
+            String cmd = info.command().orElse(null);
+            if ((cmd == null || cmd.isEmpty()) && commandLine != null) {
+                cmd = Objects.requireNonNull(commandLine);
+                String[] args = info.arguments().orElse(null);
+                if (args != null) {
+                    for (int i = args.length - 1; i >= 0; i--) {
+                        int index = cmd.lastIndexOf(args[i]);
+                        if (index >= 0) {
+                            cmd = cmd.substring(0, index);
+                        }
+                    }
+                }
+                cmd = cmd.stripTrailing();
+            }
+            command = cmd;
```

Comment:
```suggestion
    protected static record ProcessInfo(@Nullable String command, @Nullable String commandLine) {

        /**
         * Initializes the command and commandLine fields.
         * If the command field is not present, it parses the first token in the command line.
         */
        protected static ProcessInfo build(ProcessHandle.Info info) {
            @Nullable
            String commandLine = info.commandLine().orElse(null);
            String cmd = info.command().orElse(null);
            if ((cmd == null || cmd.isEmpty()) && commandLine != null) {
                cmd = Objects.requireNonNull(commandLine);
                String[] args = info.arguments().orElse(null);
                if (args != null) {
                    for (int i = args.length - 1; i >= 0; i--) {
                        int index = cmd.lastIndexOf(args[i]);
                        if (index >= 0) {
                            cmd = cmd.substring(0, index);
                        }
                    }
                }
                cmd = cmd.stripTrailing();
            }
            return new ProcessInfo(cmd, commandLine);
```

Reply:
The set contains an entry for all processes. Even those with nulls. So we just need to filter the stream before creating the set.